shell/browser/api/electron_api_app.cc

```diff
@@ -1026,7 +1041,11 @@ void App::DisableHardwareAcceleration(gin_helper::ErrorThrower thrower) {
         "before app is ready");
     return;
   }
-  content::GpuDataManager::GetInstance()->DisableHardwareAcceleration();
+  if (gpu_data_manager_initialized_) {
+    content::GpuDataManager::GetInstance()->DisableHardwareAcceleration();
```

Comment:
How come we call this in two different places...?

Reply:
`DisableHardwareAcceleration` and `DisableDomainBlockingFor3DAPIsForTesting` are apis that are called by user before `ready` event. And the time from app main script starts executing till app ready event happens is a grey area, we don't have/want to expose a signal to user that denotes `PreCreateThreads` , I am just making sure here that current api behavior is preserved while the condition for accessing `GpuDataManager` only after `PreCreateThreads` is also respected internally.